R/approach_independence.R

```diff
@@ -44,22 +60,30 @@ prepare_data.independence <- function(internal, index_features = NULL, ...) {
     ]
   }
 
+  # Convert the data.tables to matrices as all entries are numeric.
   x_train0_mat <- as.matrix(x_train0)
   x_explain0_mat <- as.matrix(x_explain0)
 
+  # Get coalition indices.
+  # We repeat each coalition index `min(n_samples, n_train)` times. We use `min`
+  # as we cannot sample `n_samples` unique indices if `n_train` is less than `n_samples`.
   index_s <- rep(seq_len(nrow(S0)), each = min(n_samples, n_train))
-  w <- 1 / n_samples # Yes, not n_samples0
-
-  n_col <- n_explain
+  w0 <- 1 / min(n_samples, n_train) # The inverse of the number of samples being used in practice
```

Comment:
I went for this for consistency. It does actually make a tiny difference what we use here, as the edge cases (full and empty set) are weighted 1, so their relative weight compared to those will differ if we choose something else.

Reply:
That makes sense, and I agree with you there. I like the way you made it in the final version.